build-and-test.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -e 	# Exit immediately upon failure
+set -o pipefail  # Carry failures over pipes
+
+repo_root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+docker_repo="microsoft/dotnet-nightly"
+
+pushd "${repo_root}" > /dev/null
+
+find . -path './.git' -prune -o -path './test' -prune -o -path './update-dependencies' -prune -o -name 'Dockerfile' -printf '%p\t%d\n' | egrep -v 'nanoserver|windowsservercore' | sort -n -k2 | while read line
```

Comment:
`printf` isn't supported by default on OS X, so I think your proposal is a preferable approach.  I was originally going to do something more like this before realizing that the default find order wasn't sufficient; I didn't think of sorting the dirnames alphabetically, though, so thanks for the suggestion.

---
In reply to: [76320023](https://github.com/dotnet/dotnet-docker-nightly/pull/35#discussion_r76320023) [](ancestors = 76320023)